In [18]:
import pandas as pd

import pprint
pp = pprint.PrettyPrinter(indent =4)

There are 9 csv files of available items that needs to be concatenate
I will first read those files and save in dictionaries

In [2]:
#Read the csv files into pandas dataframe

In [90]:
list_of_products_2000 = pd.read_csv("list_of_products_2000.csv") 
list_of_products_2000=list_of_products_2000.to_dict('records')

list_of_available_products_Rerun_2000 = pd.read_csv("list_of_available_products_Rerun_2000.csv") 
list_of_available_products_Rerun_2000=list_of_available_products_Rerun_2000.to_dict('records')

list_of_products_4000 = pd.read_csv("list_of_products_4000.csv") 
list_of_products_4000=list_of_products_4000.to_dict('records')

list_of_available_products_Rerun_4000 = pd.read_csv("list_of_available_products_Rerun_4000.csv") 
list_of_available_products_Rerun_4000=list_of_available_products_Rerun_4000.to_dict('records')

list_of_products_6000 = pd.read_csv("list_of_products_6000.csv") 
list_of_products_6000=list_of_products_6000.to_dict('records')

list_of_available_products_Rerun_6000 = pd.read_csv("list_of_available_products_Rerun_6000.csv") 
list_of_available_products_Rerun_6000=list_of_available_products_Rerun_6000.to_dict('records')

list_of_products_8000 = pd.read_csv("list_of_products_8000.csv") 
list_of_products_8000=list_of_products_8000.to_dict('records')

list_of_products_10000 = pd.read_csv("list_of_products_10000.csv") 
list_of_products_10000=list_of_products_10000.to_dict('records')

list_of_products_12000 = pd.read_csv("list_of_products_12000.csv") 
list_of_products_12000=list_of_products_12000.to_dict('records')

In [91]:
pp.pprint(list_of_products_2000[0])
pp.pprint(list_of_available_products_Rerun_2000[0])
pp.pprint(list_of_products_4000[:10])
pp.pprint(list_of_available_products_Rerun_4000[0])
pp.pprint(list_of_products_6000[:10])
pp.pprint(list_of_available_products_Rerun_6000[0])
pp.pprint(list_of_products_8000[0])
pp.pprint(list_of_products_10000[0])
pp.pprint(list_of_products_12000[0])

{   'No_of_Rating': '1,391 ratings',
    'Unnamed: 0': 0,
    'asin': '9788072216',
    'category': 'Fragrance',
    'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray '
                   'Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
    'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
    'price': '$56.29',
    'rating': '4.7 out of 5 stars'}
{   'No_of_Rating': '687 ratings',
    'Unnamed: 0': 0,
    'asin': 'B000052YQU',
    'category': 'Skin Care',
    'description': 'Special Ingredients\n'
                   '\n'
                   '\n'
                   'Keratin\n'
                   '\n'
                   '\n'
                   '\n'
                   '\n'
                   'Item Form\n'
                   '\n'
                   '\n'
                   'Cream\n'
                   '\n'
                   '\n'
                   '\n'
                   '\n'
                   'Brand\n'
                   '\n'
                

In [92]:
Final_list_of_available_products = list_of_products_2000 + list_of_available_products_Rerun_2000 + list_of_products_4000 + list_of_available_products_Rerun_4000 + list_of_products_6000 + list_of_available_products_Rerun_6000 + list_of_products_8000 + list_of_products_10000 + list_of_products_12000

In [93]:
len(Final_list_of_available_products)

4242

There are 4242 records scraped from amazon.com under the category health and beauty products
Next step is to clean the data and keep the information essential to analysis


# Data Cleaning

In [97]:
Final_list_of_available_products[:3]

[{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': '$56.29',
  'rating': '4.7 out of 5 stars',
  'No_of_Rating': '1,391 ratings'},
 {'asin': 'B00004TMFE',
  'name': 'Avalon Organics Therapy Thickening Conditioner, Biotin B-Complex, 14 Oz',
  'category': 'Hair Care',
  'description': 'Brand Avalon Organics Scent Biotin B Hair Type Thin Liquid Volume 14 Fluid Ounces Ingredients Aloe Barbadensis Leaf Juice (Certified Organic Ingredient), Aqua (Water), Brassica Alcohol, Glycerin, Cocos Nucifera (Coconut) Oil (Certified Organic Ingredient), Brassicyl Isoleucinate Esylate, Gluconolactone, Capryloyl Glycerin/Sebacic Acid Copolymer, Cedrus Atlantica Wood Oil, Citrus Aurantifolia (Lime) Oil, Citrus Aurantium Dulcis (Orange) Peel Oil, Citrus Limon (Lemon) Peel Oil, Citrus Paradisi (Grapefru

In [98]:
#Removing the column key Unnamed: 0
def new_dict(old_dict):
    n = old_dict.copy()
    n.pop('Unnamed: 0',None)
    return n

Final_list_of_available_products = list(map(new_dict,Final_list_of_available_products))

In [99]:
Final_list_of_available_products[:3]

[{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': '$56.29',
  'rating': '4.7 out of 5 stars',
  'No_of_Rating': '1,391 ratings'},
 {'asin': 'B00004TMFE',
  'name': 'Avalon Organics Therapy Thickening Conditioner, Biotin B-Complex, 14 Oz',
  'category': 'Hair Care',
  'description': 'Brand Avalon Organics Scent Biotin B Hair Type Thin Liquid Volume 14 Fluid Ounces Ingredients Aloe Barbadensis Leaf Juice (Certified Organic Ingredient), Aqua (Water), Brassica Alcohol, Glycerin, Cocos Nucifera (Coconut) Oil (Certified Organic Ingredient), Brassicyl Isoleucinate Esylate, Gluconolactone, Capryloyl Glycerin/Sebacic Acid Copolymer, Cedrus Atlantica Wood Oil, Citrus Aurantifolia (Lime) Oil, Citrus Aurantium Dulcis (Orange) Peel Oil, Citrus Limon (Lemon) Peel Oil, Citrus Paradisi (Grapefru

In [100]:
#Removing $ from price,  out of 5 stars from ratings and  ratings from No_of_ratings
def cleandata(x):
    price = x['price']
    rating = x['rating']
    No_of_Rating = x['No_of_Rating']
    x['price'] = price.replace("$","")
    x['rating'] = rating.replace(" out of 5 stars","")
    x['No_of_Rating'] = No_of_Rating.replace(" ratings","")
    x['No_of_Rating']= x['No_of_Rating'].replace(",","")
    return x

In [101]:
cleandata({'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': '$56.29',
  'rating': '4.7 out of 5 stars',
  'No_of_Rating': '1,391 ratings'})

{'asin': '9788072216',
 'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
 'category': 'Fragrance',
 'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
 'price': '56.29',
 'rating': '4.7',
 'No_of_Rating': '1391'}

In [102]:
Final_list_of_available_products = list(map(lambda x: cleandata(x), Final_list_of_available_products))

In [103]:
Final_list_of_available_products[:1]

[{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': '56.29',
  'rating': '4.7',
  'No_of_Rating': '1391'}]

There are some values for key price that is of the format
8.4-29.99. The following function filters out all the dictionaries with the format

In [104]:
def find_(x):
    price = x['price']
    return price.find('-')>0

In [109]:
list(filter(find_, Final_list_of_available_products))

[{'asin': 'B0015PGFQW',
  'name': 'TIGI Bed Head Manipulator, Texture Paste, 2 oz / 57 g',
  'category': 'Hair Care',
  'description': 'Scent\n\n\n2 PACK\n\n\n\n\nBrand\n\n\nTIGI\n\n\n\n\nItem Form\n\n\nClay',
  'price': '8.40 - 29.99',
  'rating': '4.6',
  'No_of_Rating': '61'},
 {'asin': 'B001CSNXLC',
  'name': '2 Month Supply',
  'category': 'Wall Art',
  'description': 'Brand\n\n\nKirkland Signature\n\n\n\n\nItem Weight\n\n\n0.15 Pounds\n\n\n\n\nAge Range (Description)\n\n\nAdult',
  'price': '8.25 - 76.50',
  'rating': '4.2',
  'No_of_Rating': '1067'}]

In [107]:
x =[{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': '8.40 - 29.99',
  'rating': '4.7',
  'No_of_Rating': '1391'}]

In [79]:
#changing price and rating to float and No_of ratings to string
def strtonum(x):
    price = x['price']
    rating = x['rating']
    No_of_Rating = x['No_of_Rating']
    x['price'] = float(price)
    x['rating'] = float(rating)
    x['No_of_Rating'] = int(No_of_Rating)
    return x

In [82]:
Final_list_of_available_products = list(map(lambda x: strtonum(x), Final_list_of_available_products))

ValueError: could not convert string to float: '8.40 - 29.99'

In [ ]:
#There are some values for key price that is of the format
#8.4-29.99. The following function filters out all the dictionaries with the format

In [83]:
def find_(x):
    price = x['price']
    return price.find('-')>0

In [84]:
Final_list_of_available_products[:3]

[{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': 56.29,
  'rating': 4.7,
  'No_of_Rating': 1391},
 {'asin': 'B00004TMFE',
  'name': 'Avalon Organics Therapy Thickening Conditioner, Biotin B-Complex, 14 Oz',
  'category': 'Hair Care',
  'description': 'Brand Avalon Organics Scent Biotin B Hair Type Thin Liquid Volume 14 Fluid Ounces Ingredients Aloe Barbadensis Leaf Juice (Certified Organic Ingredient), Aqua (Water), Brassica Alcohol, Glycerin, Cocos Nucifera (Coconut) Oil (Certified Organic Ingredient), Brassicyl Isoleucinate Esylate, Gluconolactone, Capryloyl Glycerin/Sebacic Acid Copolymer, Cedrus Atlantica Wood Oil, Citrus Aurantifolia (Lime) Oil, Citrus Aurantium Dulcis (Orange) Peel Oil, Citrus Limon (Lemon) Peel Oil, Citrus Paradisi (Grapefruit) Peel Oil, Eucalyptus Globul

In [ ]:
{'asin': '9788072216',
  'name': 'Prada Candy by Prada for Women 1.7 oz Eau de Parfum Spray',
  'category': 'Fragrance',
  'description': 'Brand Prada Scent Honey, Musk , Vanilla Item Form Spray Item Weight 278 Grams Item Volume 1.7 Fluid Ounces',
  'price': 56.29-60',
  'rating': 4.7,
  'No_of_Rating': 1391}

In [ ]:
find_(x)